modules/languages/bash/config.nix

```diff
@@ -0,0 +1,80 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib;
+with builtins; let
+  cfg = config.vim.languages.bash;
+  diagnostics = {
+    shellcheck = {
+      package = pkgs.shellcheck;
+      nullConfig = pkg: ''
+        table.insert(
+          ls_sources,
+          null_ls.builtins.diagnostics.shellcheck.with({
+            command = "${pkg}/bin/shellcheck",
+          })
+        )
+      '';
+    };
+  };
+
+  formats = {
+    shfmt = {
+      package = pkgs.shfmt;
+      nullConfig = ''
+        table.insert(
+          ls_sources,
+          null_ls.builtins.formatting.shfmt.with({
+            command = "${pkgs.shfmt}/bin/shfmt",
+          })
+        )
+      '';
+    };
+  };
+
+  servers = {
+    bash-ls = {
+      package = pkgs.nodePackages.bash-language-server;
+      lspConfig = ''
+        lspconfig.bashls.setup{
+          capabilities = capabilities;
+          on_attach = default_on_attach;
+          cmd = ${
+          if isList cfg.lsp.package
+          then nvim.lua.expToLua cfg.lsp.package
+          else ''{"${cfg.lsp.package}/bin/bash-language-server",  "start"}''
+        };
+        }
+      '';
+    };
+  };
```

Comment:
I don't like that these vars are duplicated in both files - it's easy to forget one when editing the other later on. maybe extract into another file and import it?

anyways, flutter module has the same problem and should be fixed as well.

Reply:
I'll probably leave fixing this to a later revision, preferably one that focuses on cleaning up the language module as a whole. Ideally we'd do something that goes roughly as

```
languages/languageName
├── commons.nix # where formatters and servers would be
├── config.nix # config
├── default.nix # imports
└── languageName.nix # option declarations
```